File: cflib_2_12/cplus/libcflib/CFLibCollisionDetectedException.cpp
/*
 *  MSS Code Factory CFLib 2.12
 *
 *	Copyright (c) 2020 Mark Stephen Sobkow
 *	
 *	Licensed under the Apache License, Version 2.0 (the "License");
 *	you may not use this file except in compliance with the License.
 *	You may obtain a copy of the License at
 *	
 *	    http://www.apache.org/licenses/LICENSE-2.0
 *	
 *	Unless required by applicable law or agreed to in writing, software
 *	distributed under the License is distributed on an "AS IS" BASIS,
 *	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *	See the License for the specific language governing permissions and
 *	limitations under the License.
 */

#include <cflib/CFLib.hpp>
#include <cflib/CFLibArgumentException.hpp>
#include <cflib/CFLibCollisionDetectedException.hpp>

namespace cflib {

	const std::string CFLibCollisionDetectedException::S_COLLISION_DETECTED( "Collision detected for primary key " );

	CFLibCollisionDetectedException::~CFLibCollisionDetectedException() {
	}

	CFLibCollisionDetectedException::CFLibCollisionDetectedException(
		const CFLibCollisionDetectedException& src )
	: std::runtime_error( src )
	{
	}

	CFLibCollisionDetectedException::CFLibCollisionDetectedException(
		const std::string& msg )
	: std::runtime_error( msg )
	{
	}

	CFLibCollisionDetectedException::CFLibCollisionDetectedException(
		const std::string& throwingClassName,
		const std::string& methName,
		const std::string& msg )
	: std::runtime_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE + msg )
	{
	}

	CFLibCollisionDetectedException::CFLibCollisionDetectedException(
		const std::string& throwingClassName,
		const std::string& methName,
		ICFLibAnyObj* argKey )
	: std::runtime_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE + S_COLLISION_DETECTED + (( argKey != NULL ) ? argKey->toString() : CFLib::S_EMPTY ))
	{
	}
}

File: cflib_2_12/java/cflib/src/org.msscf.msscf.cflib.CFLib/org/msscf/msscf/cflib/CFLib/CFLibCollisionDetectedException.java
/*
 *  MSS Code Factory CFLib 2.12
 *
 *	Copyright 2016-2026 Mark Stephen Sobkow
 *
 *	These files are part of MSS Code Factory Version 2.12.
 *
 *	MSS Code Factory is available under the terms of the GNU General Public License,
 *	Version 3 or later.
 *
 *	MSS Code Factory is free software: you can redistribute it and/or
 *	modify it under the terms of the GNU General Public License as published by
 *	the Free Software Foundation, either version 3 of the License, or
 *	(at your option) any later version.
 *
 *	MSS Code Factory is distributed in the hope that it will be useful,
 *	but WITHOUT ANY WARRANTY; without even the implied warranty of
 *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *	GNU General Public License for more details.
 */

package org.msscf.msscf.cflib.CFLib;

import java.util.*;

public class CFLibCollisionDetectedException extends CFLibRuntimeException {

	protected Object indexKey = null;

	public CFLibCollisionDetectedException(
		String msg )
	{
		super( msg );
	}

	public CFLibCollisionDetectedException(
		Class throwingClass,
		String methName,
		String msg )
	{
		super( throwingClass, methName, msg );
	}

	public CFLibCollisionDetectedException(
		Class throwingClass,
		String methName,
		String msg,
		Throwable th )
	{
		super( throwingClass, methName, msg, th );
	}

	public CFLibCollisionDetectedException(
		Class throwingClass,
		String methName,
		Object argKey )
	{
		super( throwingClass, methName,
			"Collision detected"
				+ ( ( argKey != null )
						? " for primary key " + argKey.toString()
						: "" ) );
		indexKey = argKey;
	}

	public CFLibCollisionDetectedException(
		Class throwingClass,
		String methName,
		Object argKey,
		Throwable th )
	{
		super( throwingClass, methName,
			"Collision detected"
				+ ( ( argKey != null )
						? " for primary key " + argKey.toString()
						: "" ),
			th );
		indexKey = argKey;
	}

	public Object getIndexKey() {
		return( indexKey );
	}
}
